RallyHereIntegration/Source/RallyHereEditor/Private/RallyHereEditorCustomizations.cpp

```diff
@@ -0,0 +1,283 @@
+
+	const FSimpleDelegate OnValueChanged = FSimpleDelegate::CreateLambda([&StructCustomizationUtils]()
+		{
+			StructCustomizationUtils.GetPropertyUtilities()->ForceRefresh();
+		});
+
+	for (uint32 ChildIndex = 0; ChildIndex < NumChildren; ++ChildIndex)
+	{
+		const TSharedRef< IPropertyHandle > ChildHandle = StructPropertyHandle->GetChildHandle(ChildIndex).ToSharedRef();
+
+		if (ChildHandle->GetProperty()->GetName() == TEXT("Id"))
+		{
+			IdPropertyHandle = ChildHandle;
+			ChildHandle->SetOnPropertyValueChanged(OnValueChanged);
+		}
+
+		if (ChildHandle->GetProperty()->GetName() == TEXT("LegacyId"))
+		{
+			LegacyIdPropertyHandle = ChildHandle;
+			ChildHandle->SetOnPropertyValueChanged(OnValueChanged);
+		}
+	}
+
+	check(IdPropertyHandle.IsValid());
+	check(LegacyIdPropertyHandle.IsValid());
+
+	FText Name;
+
+	TArray<void*> RawData;
+	IdPropertyHandle->AccessRawData(RawData);
+
+	FGuid Id = FGuid();
+	if (RawData.Num() > 0)
+	{
+		Id = *static_cast<FGuid*>(RawData[0]);
+	}
+
+	int32 LegacyId = 0;
+	LegacyIdPropertyHandle.Get()->GetValue(LegacyId);
+
+	TSharedRef<SHorizontalBox> ButtonArray = SNew(SHorizontalBox);
+
+	if (!Id.IsValid())
+	{
+		ButtonArray->AddSlot()
+			.AutoWidth()
+			[
+				SNew(SButton)
+				.Text(LOCTEXT("CreateItemButton", "Create"))
+				.OnClicked(this, &FRH_RHLegacyIdToGuidCustomization::OnGenerateIdClicked)
+			];
+	}
+
+	ButtonArray->AddSlot()
+	.AutoWidth()
+	[
+		SNew(SButton)
+		.Text(LOCTEXT("SetItemIdButton", "Set"))
+		.OnClicked(this, &FRH_RHLegacyIdToGuidCustomization::OnEditManuallyClicked)
+	];
+
+	HeaderRow.NameContent()
+		[
+			StructPropertyHandle->CreatePropertyNameWidget()
+		]
+	.ValueContent()
+		[
+			SNew(SVerticalBox)
+			+SVerticalBox::Slot()
+			.AutoHeight()
+			[
+				SNew(SHorizontalBox)
+				+SHorizontalBox::Slot()
+				.AutoWidth()
+				[
+					SNew(STextBlock)
+					.Text(LOCTEXT("ItemId", "Item Id:"))
+				]
+				+SHorizontalBox::Slot()
+				.AutoWidth()
+				[
+					SNew(STextBlock)
+					.Text(FText::FromString(FString::Printf(L"%s", *Id.ToString(EGuidFormats::DigitsWithHyphensLower))))
+				]
+			]
+			+SVerticalBox::Slot()
+			.AutoHeight()
+			[
+				SNew(SHorizontalBox)
+				+SHorizontalBox::Slot()
+				.AutoWidth()
+				[
+					SNew(STextBlock)
+					.Text(LOCTEXT("LegacyItemId", "Legacy Item Id:"))
+				]
+				+SHorizontalBox::Slot()
+				.AutoWidth()
+				[
+					SNew(STextBlock)
+					.Text(FText::FromString(FString::Printf(L"%d", LegacyId)))
+				]
+			]
+			+SVerticalBox::Slot()
+			.AutoHeight()
+			[
+				ButtonArray
+			]
+
+		];
+}
+
+FReply FRH_RHLegacyIdToGuidCustomization::OnGenerateIdClicked()
+{
+	// Check if we are logged in before prompting to make this call, if we are not logged in, promt for getting a new login access token.
+	auto AuthContext = FModuleManager::Get().LoadModuleChecked<FRallyHereEditorModule>(FRallyHereEditorModule::GetModuleName()).GetAuthContext();
+	if (!AuthContext->IsLoggedIn())
+	{
+		AuthContext->Refresh();
```

Comment:
Does auth context refresh pop a login box?

Reply:
AuthContext Refresh triggers a delegate that leads to the login window popping